File: BackEnd/Proyecto2_SS24/src/main/java/com/mycompany/proyecto2_ss24/backend/model/anuncios/AnuncioVideo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.proyecto2_ss24.backend.model.anuncios;

import java.io.InputStream;

/**
 *
 * @author Carlos Cotom
 */
public class AnuncioVideo extends Anuncio {
    
    private InputStream video;
    private int idAnuncioVideo;

    public AnuncioVideo() {
    }

    public AnuncioVideo(double precio, int vigenciaDias, boolean isActivo, int idInversionista, int idPeriodoTiempo, int idTipoAnuncio) {
        super(precio, vigenciaDias, isActivo, idInversionista, idPeriodoTiempo, idTipoAnuncio);
    }

    public AnuncioVideo(InputStream video, double precio, int vigenciaDias, boolean isActivo, int idInversionista, int idPeriodoTiempo, int idTipoAnuncio) {
        super(precio, vigenciaDias, isActivo, idInversionista, idPeriodoTiempo, idTipoAnuncio);
        this.video = video;
    }

    public InputStream getVideo() {
        return video;
    }

    public void setVideo(InputStream video) {
        this.video = video;
    }    

    public int getIdAnuncioVideo() {
        return idAnuncioVideo;
    }

    public void setIdAnuncioVideo(int idAnuncioVideo) {
        this.idAnuncioVideo = idAnuncioVideo;
    }
    
}

File: FrontEnd/Proyecto2_SS24/src/app/models/anuncioVideo.ts
export class AnuncioVideo {

    precio: number;
    vigenciaDias: number;
    isActivo: boolean;
    idInversionista: number;
    idPeriodoTiempo: number;
    idTipoAnuncio: number;
    idAnuncio: number;
    titulo: string;
    idAnuncioTexto: number;
    fechaCompra: string

    constructor(precio: number, vigenciaDias: number, isActivo: boolean,
        idInversionista: number, idPeriodoTiempo: number, idTipoAnuncio: number,
        idAnuncio: number, titulo: string, idAnuncioTexto: number,fechaCompra: string
    ) {
        this.precio = precio;
        this.vigenciaDias = vigenciaDias;
        this.isActivo = isActivo;
        this.idInversionista = idInversionista;
        this.idPeriodoTiempo = idPeriodoTiempo;
        this.idTipoAnuncio = idTipoAnuncio;
        this.idAnuncio = idAnuncio;
        this.titulo = titulo;
        this.idAnuncioTexto = idAnuncioTexto;
        this.fechaCompra = fechaCompra;
    }

}